domino/datasets.py

```diff
@@ -0,0 +1,174 @@
+import hashlib
+import math
+import os
+import time
+from concurrent.futures import ThreadPoolExecutor
+from logging import Logger
+from typing import AnyStr
+
+from domino.http_request_manager import _HttpRequestManager
+from domino.routes import _Routes
+
+FILE_UPLOAD_SETTING_DEFAULT = "Overwrite"
+MAX_WORKERS = 10
+MAX_UPLOAD_ATTEMPTS = 10
+MB = 2 ** 20  # 2^20 bytes - 1 Megabyte
+SLEEP_TIME_IN_SEC = 5
+UPLOAD_READ_TIMEOUT_IN_SEC = 30
+
+
+class UploadChunk:
```

Comment:
seems to be a good use case for a dataclass https://docs.python.org/3/library/dataclasses.html